tests/integration/targets/setup_iis/handlers/restore_iis_conf.yml

```
@@ -0,0 +1,24 @@
+- name: stop IIS service
+  ansible.windows.win_service:
+    name: W3SVC
+    state: stopped
+
+- name: sleep for 10 seconds after stopping IIS service
+  pause:
+    seconds: 10
+
+- name: restore iis configuration file
+  ansible.windows.win_copy:
+    src: '{{ iis_config_file_path }}\{{ iis_config_bck_file_name }}'
+    dest: '{{ iis_config_file_path }}\{{ iis_config_file_name }}'
+    remote_src: yes
```

Instead of sleeping let's try and add a retry to the task.

```suggestion
- name: restore iis configuration file
  ansible.windows.win_copy:
    src: '{{ iis_config_file_path }}\{{ iis_config_bck_file_name }}'
    dest: '{{ iis_config_file_path }}\{{ iis_config_file_name }}'
    remote_src: yes
  register: copy_result
  until: copy_result is successful
  retries: 5
  delay: 5
```